site/_posts/2020-05-22-building-cli-app-with-device-grant-and-golang.md

```diff
@@ -0,0 +1,303 @@
+
+If you're a bit hazy on the Device Code grant, then here is a quick refresher from the [grant's RFC (#8628)](https://tools.ietf.org/html/rfc8628):
+
+> The OAuth 2.0 device authorization grant is designed for Internet-
+   connected devices that either lack a browser to perform a user-agent-
+   based authorization or are input constrained to the extent that
+   requiring the user to input text in order to authenticate during the
+   authorization flow is impractical.  It enables OAuth clients on such
+   devices (like smart TVs, media consoles, digital picture frames, and
+   printers) to obtain user authorization to access protected resources
+   by using a user agent on a separate device.
+
+![Device Code Grant Diagram](https://is.docs.wso2.com/en/5.11.0/assets/img/using-wso2-identity-server/deviceflow.png)
+
+## Setup Instructions
+
+First things first, to follow along you'll need golang obviously. If you don't already have golang on your local machine, *go* ahead and [install it over here](https://golang.org/doc/install).
+
+Next, we need to set up FusionAuth (FA) on our local machine to properly be able to test our CLI app. If you haven't already played around with FA then I suggest checking out their [5 minute setup guide](https://fusionauth.io/docs/v1/tech/5-minute-setup-guide) and stopping after you've completed step #3. That should get you set up with a running FA server with which our CLI app can authenticate.
+
+Once you've got FA running, we'll need to create our CLI app's FA Application. Follow these steps to do so:
+
+1. Head over to [FA's Applications UI](http://localhost:9011/admin/application/)
+1. Click the green "➕" button to add a new Application.
+1. Give your Application a good name so you remember what it is. "Device Code Grant Example" isn't a bad choice.
+1. Click on the OAuth tab and under "Enabled grants", check off "Device". This will enable support for the Device Code Grant type on our newly created Application.
+1. Since it is required, be sure to fill in the "Device verification URL" field. You can just put "https://example.com/device" if you'd like since we don't need this for our particular application.
+   1. Since our CLI app is actually running from a device with a browser, we don't need to provide a friendly URL to the user as we can open FA's `/oauth2/device` URL directly. Neat!
+1. Your application form should look something like the below, and if so then you're doing great! Click the save button in the top right.
+{% include _image.liquid src="/assets/img/blogs/golang-cli-device-grant/fa-create-application-example.png" alt="FA Create Application Example." class="img-fluid" figure=false %}
```

Comment:
This image breaks the list numbering. I think it needs to be indented so that Markdown considers it part of the <ol>

Reply:
Thanks, fixed.